### PR TITLE
[qttools] fix qdbusviewer

### PR DESCRIPTION
**Pull request by reitowo**
This will fix building `qttools` on macOS that previously the portfile will try to copy `qdbusviewer.app`, but if the `qtbase[dbus]` is not installed, this tool will not be built, result in a build failure (copy and file not found).

By investigating Windows build, the `qdbusviewer.exe` is also missing if `qtbase[dbus]` is not enabled. So this port is "mis-compiling" `qdbusviewer` at all, probably because previous maintainer just enabled `qtbase[dbus]` in other dependency and considering `qdbusviewer` is built by `qttools` as a default (which is not).

Regarding that, create a new feature `qttools[qdbusviewer]` to add `qtbase[dbus]` as a dependency and better control of this build behaviour, and conditionally copy `qdbusviewer.app` if it is actually built.

### CODE REVIEW

File: ports/qttools/portfile.cmake [hunk 1]
```diff
@@ -90,7 +90,11 @@ qt_install_submodule(PATCHES    ${${PORT}_PATCHES}
                     )
 
 if(VCPKG_TARGET_IS_OSX)
-    set(OSX_APP_FOLDERS Designer.app Linguist.app pixeltool.app qdbusviewer.app)
+    set(OSX_APP_FOLDERS Designer.app Linguist.app pixeltool.app)
+    if ("qdbusviewer" IN_LIST FEATURES)
+        message(STATUS "Built qdbusviewer")
+        set(OSX_APP_FOLDERS "${OSX_APP_FOLDERS}" qdbusviewer.app)
+    endif()
```
Comment (Neumann-A) [inline ports/qttools/portfile.cmake:97]: this doesn't deal with the case that `qtbase` was build with `dbus` but it wasn't requested by `qttools`. You probably need to control a `FEATURE_` option

Comment (cnSchwarzer) [reply ports/qttools/portfile.cmake:97]: I tried, and the vcpkg warned me that FEATURE_qdbusviewer is not used by cmake configure.

Comment (cnSchwarzer) [reply ports/qttools/portfile.cmake:97]: I think we can just file(GLOB...) the build folder? There's no point to pre-create a list.

Comment (Neumann-A) [reply ports/qttools/portfile.cmake:97]: you need to check the actual sources to see how the feautre is named: https://github.com/qt/qttools/blob/c1d40191c672dcf68d658b434085d3288bda97eb/configure.cmake#L78
so try `FEATURE_qdbus`

File: ports/qttools/vcpkg.json [hunk 2]
```diff
@@ -59,6 +60,18 @@
...
       ]
     },
+    "qdbusviewer": {
+      "description": "Build qdbusviewer",
+      "dependencies": [
+        {
+          "name": "qtbase",
+          "default-features": false,
+          "features": [
+            "dbus"
```
Comment (Neumann-A) [inline ports/qttools/vcpkg.json:70]: this also depends on `widgets`.

File: ports/qttools/vcpkg.json [hunk 3]
```diff
@@ -59,6 +60,18 @@
         }
       ]
     },
+    "qdbusviewer": {
```
Comment (Neumann-A) [inline ports/qttools/vcpkg.json:63]: maybe this should only be named qdbus according to the sources

**Comment by BillyONeal** [comment]
I would like @Neumann-A to affirm that he is happy with any changes to qt's guts

File: ports/qttools/portfile.cmake [hunk 4]
```diff
@@ -90,7 +91,11 @@ qt_install_submodule(PATCHES    ${${PORT}_PATCHES}
                     )
 
 if(VCPKG_TARGET_IS_OSX)
-    set(OSX_APP_FOLDERS Designer.app Linguist.app pixeltool.app qdbusviewer.app)
+    set(OSX_APP_FOLDERS Designer.app Linguist.app pixeltool.app)
+    if (FEATURE_qdbus)
+        message(STATUS "Built qdbusviewer")
+        set(OSX_APP_FOLDERS "${OSX_APP_FOLDERS}" qdbusviewer.app)
```
Comment (Neumann-A) [inline ports/qttools/portfile.cmake:97]:
```suggestion
        list(APPEND OSX_APP_FOLDERS qdbusviewer.app)
```

**Review by Neumann-A — CHANGES_REQUESTED** [review]
the little list change and then it should be ok.

**Comment by BillyONeal** [comment]
Thanks for the fix @cnSchwarzer and thanks for the review @Neumann-A !

**Pull request merged**